dimension/client.go

```diff
@@ -55,36 +56,76 @@ func (c *Client) Checker(ctx context.Context, check *healthcheck.CheckState) err
 	return c.hcCli.Checker(ctx, check)
 }
 
-// GetAreaTypes retrieves the Cantabular area-types associated with a dataset
-func (c *Client) GetAreaTypes(ctx context.Context, userAuthToken, serviceAuthToken, datasetID string) (GetAreaTypesResponse, error) {
-	logData := log.Data{
-		"method":     http.MethodGet,
-		"dataset_id": datasetID,
+func (c *Client) createDimensionGetRequest(ctx context.Context, userAuthToken, serviceAuthToken, urlPath string, logData log.Data, urlValues url.Values) (*http.Request, error) {
+	areasURL, err := c.baseURL.Parse(urlPath)
+	if err != nil {
+		return &http.Request{}, dperrors.New(
+			errors.Wrap(err, "failed to parse areas URL"),
+			http.StatusInternalServerError,
+			logData,
+		)
 	}
 
-	areaTypesURL, err := c.baseURL.Parse("area-types")
+	areasURL.RawQuery = urlValues.Encode()
+	reqURL := areasURL.String()
+
+	req, err := newRequest(ctx, http.MethodGet, reqURL, nil, userAuthToken, serviceAuthToken)
 	if err != nil {
-		return GetAreaTypesResponse{}, dperrors.New(
-			fmt.Errorf("failed to parse area-types URL: %w", err),
+		return &http.Request{}, dperrors.New(
+			errors.Wrap(err, "failed to create request"),
+			http.StatusBadRequest,
+			logData,
+		)
+	}
+	return req, nil
+}
+
+func checkDimensionGetResponse(logData log.Data, resp http.Response) error {
+	if resp.StatusCode == http.StatusNotFound {
+		var errorResp ErrorResp
+		if err := json.NewDecoder(resp.Body).Decode(&errorResp); err == nil {
+			return dperrors.New(
+				fmt.Errorf("error response from Dimensions API (%d): %w", resp.StatusCode, errorResp),
+				http.StatusInternalServerError,
+				logData,
+			)
+		}
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		// Best effort — an empty body is fine for the error message
+		body, _ := io.ReadAll(resp.Body)
+		return dperrors.New(
+			fmt.Errorf("error response from Dimensions API (%d): %s", resp.StatusCode, body),
 			http.StatusInternalServerError,
 			logData,
 		)
 	}
 
-	areaTypesURL.RawQuery = url.Values{"dataset": []string{datasetID}}.Encode()
-	reqURL := areaTypesURL.String()
+	return nil
+}
 
-	clientlog.Do(ctx, "getting area types", service, reqURL, logData)
+// GetAreaTypes retrieves the Cantabular area-types associated with a dataset
+func (c *Client) GetAreaTypes(ctx context.Context, userAuthToken, serviceAuthToken, datasetID string) (GetAreaTypesResponse, error) {
```

Comment:
I think it would be better to use an input struct for these functions rather than long lists of arguments
`GetAreaTypes(ctx context.Context, input GetAreaTypesInput)`
This way it will be easier to change in the future without breaking (and will be tidier too)

Reply:
I've made this change for `GetAreas` but `GetAreaTypes` was not added in this PR and is actually already used so changing it now would break the frontend.